airbyte-cdk/python/airbyte_cdk/sources/streams/core.py

```diff
@@ -111,7 +111,10 @@ def get_json_schema(self) -> Mapping[str, Any]:
 
     def as_airbyte_stream(self) -> AirbyteStream:
         stream = AirbyteStream(name=self.name, json_schema=dict(self.get_json_schema()), supported_sync_modes=[SyncMode.full_refresh])
-
+        
+        if self.namespace and self.namespace != "":
```

Comment:
```suggestion
        if self.namespace:
```
`bool("")` returns `False` so `if self.namespace` is enough.